### PR TITLE
Python 3.5 support, and Travis improvements

### PR DESCRIPTION
**Pull request by pjdelport**
This adds Python 3.5 to the list of tested and supported environments.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,5 +1,6 @@
 language: python
-python: 2.7
+# Currently, Travis only makes Python 3.5 available if it's the version declared here.
+python: 3.5
```
Comment (econchick) [inline .travis.yml]: Can we do something like [this](https://github.com/hynek/attrs/blob/master/.travis.yml#L9-L32)? I've been meaning to do this, as it gives a better visual within travis too:

Current (from this PR):
2016-02-15 at 12 13 pm

Proposed (from the attrs repo):
2016-02-15 at 12 12 pm

**Comment by econchick** [comment]
I've heard that the mocks don't really work with Python 3.5 (see [this conversation](https://github.com/spotify/ramlfications/pull/52#issuecomment-168653903)) - I've been meaning to look deeper into this, I can't remember what the issue is exactly. And I'll be frank - mocking is still a bit of a mystery to me, I could have just been mocking incorrectly this whole time. I did find it difficult to have mocks for both with and without `requests` installed.

I'd be okay if you want to put in some [skipif](http://pytest.org/latest/skipping.html) statements where mock fails just to see if the rest of the code works with 3.5 (and continue to add to the technical debt for me to check up on when I can :D)

**Comment by pjdelport** [comment]
I think those tests are actually failing due to the problem described in this article:

"[assert_called_once: Threat or Menace](http://engineeringblog.yelp.com/2015/02/assert_called_once-threat-or-menace.html)" (Tyler Roscoe)

In short, `assert_called_once()` is not a valid assertion method, so the mock object treats it as a mocked method instead, and silently does nothing!

The right method to use instead is `assert_called_once_with()`.

Python 3.5 and the newer mock library has a heuristic check which detects this, which what the Python3.5 build now fails with.

I started a branch to fix this problem.

**Comment by econchick** [comment]
@pjdelport you're awesome!

**Comment by pjdelport** [comment]
Okay, once #89 is merged, this should pass as well. :)

**Comment by pjdelport** [comment]
@econchick: The way attrs does it is nice!

**Comment by econchick** [comment]
Merging this on mobile 😜 here's hoping this works!

**Pull request merged**